plugins/woocommerce/src/Internal/Admin/Logging/FileV2/ListTable.php

```diff
@@ -0,0 +1,319 @@
+<?php
+declare( strict_types = 1 );
+
+namespace Automattic\WooCommerce\Internal\Admin\Logging\FileV2;
+
+use Automattic\WooCommerce\Internal\Admin\Logging\PageController;
+
+use WP_List_Table;
+
+/**
+ * ListTable class.
+ */
+class ListTable extends WP_List_Table {
+	/**
+	 * The user option key for saving the preferred number of files displayed per page.
+	 *
+	 * @const string
+	 */
+	public const PER_PAGE_USER_OPTION_KEY = 'woocommerce_logging_file_list_per_page';
+
+	/**
+	 * Instance of FileController.
+	 *
+	 * @var FileController
+	 */
+	private $file_controller;
+
+	/**
+	 * Instance of PageController.
+	 *
+	 * @var PageController
+	 */
+	private $page_controller;
+
+	/**
+	 * ListTable class.
+	 *
+	 * @param FileController $file_controller Instance of FileController.
+	 * @param PageController $page_controller Instance of PageController.
+	 */
+	public function __construct( FileController $file_controller, PageController $page_controller ) {
+		$this->file_controller = $file_controller;
+		$this->page_controller = $page_controller;
+
+		parent::__construct(
+			array(
+				'singular' => 'log-file',
+				'plural'   => 'log-files',
+				'ajax'     => false,
+			)
+		);
+	}
+
+	/**
+	 * Render message when there are no items.
+	 *
+	 * @return void
+	 */
+	public function no_items(): void {
+		esc_html_e( 'No log files found.', 'woocommerce' );
+	}
+
+	/**
+	 * Retrieves the list of bulk actions available for this table.
+	 *
+	 * @return array
+	 */
+	protected function get_bulk_actions(): array {
+		return array(
+			'delete' => __( 'Delete permanently', 'woocommerce' ),
+		);
+	}
+
+	/**
+	 * Get the existing log sources for the filter dropdown.
+	 *
+	 * @return array
+	 */
+	protected function get_sources_list(): array {
+		$sources = $this->file_controller->get_file_sources();
+		if ( is_wp_error( $sources ) ) {
+			return array();
+		}
+
+		sort( $sources );
+
+		return $sources;
+	}
+
+	/**
+	 * Displays extra controls between bulk actions and pagination.
+	 *
+	 * @param string $which The location of the tablenav being rendered. 'top' or 'bottom'.
+	 *
+	 * @return void
+	 */
+	protected function extra_tablenav( $which ): void {
+		$all_sources    = $this->get_sources_list();
+		$current_source = filter_input( INPUT_GET, 'source', FILTER_SANITIZE_STRING ) ?? '';
```

Comment:
```suggestion
		$current_source = sanitize_text_field( wp_unslash( $_GET['source'] ?? '' ) );
```

☝🏼 Should we do something like this? `FILTER_SANITIZE_STRING` is deprecated as of PHP 8.1 (in favor of `htmlspecialchars()`, which is another option).

Reply:
Or else just:

```suggestion
		$current_source = filter_input( INPUT_GET, 'source' ) ?? '';
```

Reply:
> deprecated as of PHP 8.1

Ah shoot. When I read [the docs](https://www.php.net/manual/en/filter.filters.sanitize.php) I originally interpreted them as saying that `FILTER_FLAG_NO_ENCODE_QUOTES` was being deprecated, but upon further review, I think you're right. I'm pretty sure phpcs won't like the second one, so the first suggestion is probably best.

Reply:
Ended up going with `sanitize_file_name` (a WP Core function) in 79640078d4

Reply:
(Also found two other usages of `FILTER_SANITIZE_STRING` to get rid of)